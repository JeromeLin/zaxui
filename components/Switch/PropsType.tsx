export default interface PropsType {
  theme?: 'default' | 'primary' | 'info' | 'success' | 'warning' | 'error';
  checked?: boolean;
  defaultChecked?: boolean;
  disabled?: boolean;
  onChange?: (checked: boolean) => void;
}
