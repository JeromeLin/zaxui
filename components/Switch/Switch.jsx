import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

function getChecked(props, defaultChecked) {
  if ('checked' in props) {
    return props.checked;
  }
  if ('defaultChecked' in props) {
    return props.defaultChecked;
  }
  return defaultChecked;
}

class Switch extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      checked: getChecked(props, false),
    };
  }

  componentWillReceiveProps(nextProps) {
    if ('checked' in nextProps && nextProps.checked !== this.props.checked) {
      this.setState({
        checked: getChecked(nextProps, false),
      });
    }
  }

  _onClick() {
    const { onChange } = this.props;
    const checked = !this.state.checked;
    this.setState({
      checked,
    });
    typeof onChange === 'function' && onChange(checked);
  }

  render() {
    const { prefixCls, className, theme, size, disabled } = this.props;
    const { checked } = this.state;

    const cls = classnames({
      [`${prefixCls}`]: true,
      [className]: !!className,
      [`theme-${theme}`]: !!theme,
      [`size-${size}`]: !!size,
      checked,
      disabled,
    });

    return (
      <span className={cls}>
        <input type="checkbox" className={`${prefixCls}-input`} disabled={disabled} checked={checked} onChange={() => !disabled && this._onClick(checked)} />
      </span>
    );
  }
}

Switch.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  theme: PropTypes.oneOf(['default', 'primary', 'info', 'success', 'warning', 'error']),
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
};

Switch.defaultProps = {
  prefixCls: 'za-switch',
  className: null,
  theme: 'primary',
  size: null,
  disabled: false,
  onChange() {},
};

export default Switch;
