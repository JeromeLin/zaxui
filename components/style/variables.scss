
/* base */
$theme-default: rgb(19, 17, 17);
$theme-default-light: rgba($theme-default, 0.1);
$theme-default-dark: darken($theme-default, 8%);

$theme-primary: #12c287;
$theme-primary-light: rgba($theme-primary, 0.1);
$theme-primary-dark: darken($theme-primary, 8%);

$theme-success: #69c964;
$theme-success-light: rgba($theme-success, 0.1);
$theme-success-dark: darken($theme-success, 8%);

$theme-warning: #eca632;
$theme-warning-light: rgba($theme-warning, 0.1);
$theme-warning-dark: darken($theme-warning, 8%);

$theme-error: #e55546;
$theme-error-light: rgba($theme-error, 0.1);
$theme-error-dark: darken($theme-error, 8%);

$radius-lg: r(4);
$radius-md: r(4);
$radius-sm: r(4);

$height-lg: r(52);
$height-md: r(45);
$height-sm: r(27);

$padding-lg: r(25);
$padding-md: r(16);
$padding-sm: r(10);

$fontSize-lg: r(20);
$fontSize-md: r(18);
$fontSize-sm: r(14);

$zIndex-mask: 1000;
$zIndex-popup: 1300;
$zIndex-modal: 1400;
$zIndex-toast: 1500;
$zIndex-loading: 1500;

$disabled-opacity: .5;
$disabled-background: #e1e1e1;
$disabled-color: #999;
$placeholder-color: #a9a9a9;
$icon-url: '//at.alicdn.com/t/font_126288_uipjvv8ogdivbo6r';

/* spinner */
$spinner-diameter-md: r(20);
$spinner-diameter-lg: r(31);

/* prefixCls */
$prefixCls-actionsheet: za-actionsheet;
$prefixCls-alert: za-alert;
$prefixCls-badge: za-badge;
$prefixCls-button: za-button;
$prefixCls-cell: za-cell;
$prefixCls-checkbox: za-checkbox;
$prefixCls-confirm: za-confirm;
$prefixCls-icon: za-icon;
$prefixCls-input: za-input;
$prefixCls-keyboard: za-keyboard;
$prefixCls-loading: za-loading;
$prefixCls-mask: za-mask;
$prefixCls-message: za-message;
$prefixCls-modal: za-modal;
$prefixCls-noticebar: za-noticebar;
$prefixCls-panel: za-panel;
$prefixCls-picker: za-picker;
$prefixCls-popup: za-popup;
$prefixCls-progress: za-progress;
$prefixCls-pull: za-pull;
$prefixCls-radio: za-radio;
$prefixCls-slider: za-slider;
$prefixCls-spinner: za-spinner;
$prefixCls-stepper: za-stepper;
$prefixCls-swipe: za-swipe;
$prefixCls-swipeaction: za-swipeaction;
$prefixCls-switch: za-switch;
$prefixCls-tab: za-tab;
$prefixCls-toast: za-toast;
$prefixCls-tooltip: za-tooltip;
$prefixCls-uploader: za-uploader;

/* button */

// $button-default-color: $theme-default;
// $button-default-color--active: $theme-default;
// $button-default-color--bordered: $theme-default;
// $button-default-background: #fff;
// $button-default-background--active: #ddd;
// $button-default-border: #ddd;
// $button-default-border--active: #ddd;

$button-default-color: $theme-default;
$button-default-color--active: $theme-default;
$button-default-color--bordered: $theme-default;
$button-default-background: #ddd;
$button-default-background--active: #c4c4c4;
$button-default-border: #ddd;
$button-default-border--active: #c4c4c4;

$button-primary-color: #fff;
$button-primary-color--active: rgba(#fff, .3);
$button-primary-color--bordered: $theme-primary;
$button-primary-background: $theme-primary;
$button-primary-background--active: $theme-primary-dark;
$button-primary-border: $theme-primary;
$button-primary-border--active: $theme-primary-dark;

$button-success-color: #fff;
$button-success-color--active: rgba(#fff, .3);
$button-success-color--bordered: $theme-success;
$button-success-background: $theme-success;
$button-success-background--active: $theme-success-dark;
$button-success-border: $theme-success;
$button-success-border--active: $theme-success-dark;

$button-warning-color: #fff;
$button-warning-color--active: rgba(#fff, .3);
$button-warning-color--bordered: $theme-warning;
$button-warning-background: $theme-warning;
$button-warning-background--active: $theme-warning-dark;
$button-warning-border: $theme-warning;
$button-warning-border--active: $theme-warning-dark;

$button-error-color: #fff;
$button-error-color--active: rgba(#fff, .3);
$button-error-color--bordered: $theme-error;
$button-error-background: $theme-error;
$button-error-background--active: $theme-error-dark;
$button-error-border: $theme-error;
$button-error-border--active: $theme-error-dark;