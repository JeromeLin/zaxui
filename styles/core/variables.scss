
/* base */
$base-fontSize: 14;

$theme-default: #ddd;
$theme-default-light: rgba($theme-default, 0.5);
$theme-default-dark: darken($theme-default, 10%);

$theme-info: #12c287;
$theme-info-light: rgba($theme-info, 0.1);
$theme-info-dark: darken($theme-info, 10%);

$theme-success: #69c964;
$theme-success-light: rgba($theme-success, 0.1);
$theme-success-dark: darken($theme-success, 10%);

$theme-warning: #eca632;
$theme-warning-light: rgba($theme-warning, 0.1);
$theme-warning-dark: darken($theme-warning, 10%);

$theme-error: #e55546;
$theme-error-light: rgba($theme-error, 0.1);
$theme-error-dark: darken($theme-error, 10%);

$radius-xl: r(4);
$radius-lg: r(4);
$radius-md: r(4);
$radius-sm: r(4);
$radius-xs: r(4);

$height-xl: r(52);
$height-lg: r(48);
$height-md: r(44);
$height-sm: r(34);
$height-xs: r(27);

$padding-xl: r(25);
$padding-lg: r(20);
$padding-md: r(16);
$padding-sm: r(15);
$padding-xs: r(10);

$fontSize-xl: r(18);
$fontSize-lg: r(18);
$fontSize-md: r(17);
$fontSize-sm: r(14);
$fontSize-xs: r(14);

$zIndex-mask: 1000;
$zIndex-popup: 1300;
$zIndex-modal: 1400;
$zIndex-toast: 1500;
$zIndex-loading: 1500;

$disabled-opacity: .5;
$disabled-background: #ddd;
$disabled-color: #bbb;
$placeholder-color: #a9a9a9;
$icon-url: '//at.alicdn.com/t/font_1rke47h9f6r';

/* button */
$button-default-color: #333;
$button-default-bg: $theme-default;
$button-default-border: $theme-default;
$button-default-activebg: $theme-default-light;

$button-info-color: #fff;
$button-info-bg: $theme-info;
$button-info-border: $theme-info;
$button-info-activebg: $theme-info-light;

$button-success-color: #fff;
$button-success-bg: $theme-success;
$button-success-border: $theme-success;
$button-success-activebg: $theme-success-light;

$button-warning-color: #fff;
$button-warning-bg: $theme-warning;
$button-warning-border: $theme-warning;
$button-warning-activebg: $theme-warning-light;

$button-error-color: #fff;
$button-error-bg: $theme-error;
$button-error-border: $theme-error;
$button-error-activebg: $theme-error-light;

$screen-xs:                 480px !default;

$screen-xs-min:             $screen-xs !default;
$screen-phone:              $screen-xs-min !default;
$screen-sm:                 768px !default;
$screen-sm-min:             $screen-sm !default;
$screen-tablet:             $screen-sm-min !default;

$screen-md:                 992px !default;
$screen-md-min:             $screen-md !default;
$screen-desktop:            $screen-md-min !default;

$screen-lg:                 1200px !default;
$screen-lg-min:             $screen-lg !default;
$screen-lg-desktop:         $screen-lg-min !default;

$screen-xs-max:             ($screen-sm-min - 1) !default;
$screen-sm-max:             ($screen-md-min - 1) !default;
$screen-md-max:             ($screen-lg-min - 1) !default;

$grid-columns:              12 !default;
$grid-gutter-width:         30px !default;
$grid-float-breakpoint:     $screen-sm-min !default;
$grid-float-breakpoint-max: ($grid-float-breakpoint - 1) !default;

$container-tablet:          (750px + $grid-gutter-width) !default;
$container-sm:              $container-tablet !default;

$container-desktop:         (970px + $grid-gutter-width) !default;
$container-md:              $container-desktop !default;

$container-large-desktop:   (1170px + $grid-gutter-width) !default;
$container-lg:              $container-large-desktop !default;
