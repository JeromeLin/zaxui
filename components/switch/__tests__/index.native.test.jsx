import React from 'react';
import { PixelRatio, TouchableWithoutFeedback } from 'react-native';
import { render, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Switch from '../index.native';
import variables from '../../style/themes/default.native';

describe('Switch', () => {
  it('renders correctly', () => {
    const wrapper = render(<Switch />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('defaultChecked', () => {
    const wrapper = render(<Switch defaultChecked />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('onChange', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Switch checked onChange={onChange} />);
    wrapper.find(TouchableWithoutFeedback).simulate('press', false);
    wrapper.find(TouchableWithoutFeedback).simulate('press', true);
    expect(onChange).toBeCalledWith(true);
  });

  it('disabled', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Switch checked disabled onChange={onChange} />);
    wrapper.find(TouchableWithoutFeedback).simulate('press', false);
    expect(wrapper.state('checked')).toBe(true);
  });

  it('Switch change styles', () => {
    const styles = {
      wrapperStyle: {
        alignItems: 'center',
        justifyContent: 'center',
        width: variables.switch_width,
        height: variables.switch_height,
        borderRadius: variables.switch_height / 2,
        backgroundColor: variables.switch_background,
      },
      disabledWrapperStyle: {
        opacity: 0.5,
      },
      wrapperInActive: {
        borderWidth: 3 / PixelRatio.get(),
        borderColor: variables.switch_border_color,
        backgroundColor: '#ffffff',
      },
      wrapperActive: {
        borderWidth: 0,
        backgroundColor: '#12C287',
      },
      circleStyle: {
        width: 29,
        height: 29,
        borderRadius: 15,
        backgroundColor: '#ffffff',
      },
      circleInActive: {
        shadowOffset: { width: 0, height: 0 },
        shadowOpacity: 1,
        shadowRadius: 2,
        shadowColor: 'rgba(0,0,0,0.2)',
      },
      circleActive: {
        width: 28,
        height: 27,
        shadowOffset: { width: 0, height: 0 },
        shadowOpacity: 1,
        shadowRadius: 2,
        shadowColor: 'rgba(0,127,84,1)',
      },
    };
    const wrapper = shallow(<Switch styles={styles} />);
    wrapper.setProps({ checked: true });
  });
});
