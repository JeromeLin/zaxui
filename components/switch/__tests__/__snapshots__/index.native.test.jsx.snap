// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch defaultChecked 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock />
  </react-native-mock>
</react-native-mock>
`;

exports[`Switch renders correctly 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock />
  </react-native-mock>
</react-native-mock>
`;
